```
Error running mint automation
+ mkdir -p 8340-68a4484/gopath/src/github.com/minio
+ git -C 8340-68a4484/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 8340-68a4484/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 8340-68a4484/gopath/src/github.com/minio/minio fetch --quiet minio pull/8340/head:pr8340
+ git -C 8340-68a4484/gopath/src/github.com/minio/minio checkout --quiet pr8340
+ GO111MODULE=on
+ GOPATH=/mnt/nvme/mint-auto/8340-68a4484/gopath
+ make -C 8340-68a4484/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/cli v1.21.0
go: downloading github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: downloading github.com/json-iterator/go v1.1.7
go: downloading github.com/Azure/go-autorest v11.7.0+incompatible
go: downloading github.com/minio/gokrb5/v7 v7.2.5
go: downloading google.golang.org/api v0.4.0
go: downloading github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: downloading github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: extracting github.com/json-iterator/go v1.1.7
go: extracting github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: downloading gopkg.in/ldap.v3 v3.0.3
go: extracting github.com/minio/cli v1.21.0
go: downloading github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: extracting gopkg.in/ldap.v3 v3.0.3
go: extracting github.com/Azure/go-autorest v11.7.0+incompatible
go: downloading github.com/gorilla/rpc v1.2.0
go: downloading github.com/rs/cors v1.6.0
go: extracting github.com/rs/cors v1.6.0
go: downloading github.com/minio/highwayhash v1.0.0
go: extracting github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: extracting github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: extracting gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading github.com/minio/minio-go/v6 v6.0.37
go: downloading github.com/gorilla/handlers v1.4.0
go: downloading github.com/ncw/directio v1.0.5
go: extracting github.com/gorilla/handlers v1.4.0
go: downloading go.uber.org/atomic v1.3.2
go: extracting go.uber.org/atomic v1.3.2
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/ncw/directio v1.0.5
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/minio/highwayhash v1.0.0
go: downloading github.com/miekg/dns v1.1.8
go: extracting github.com/gorilla/rpc v1.2.0
go: downloading github.com/klauspost/reedsolomon v1.9.3
go: extracting github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/minio/minio-go/v6 v6.0.37
go: downloading golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: extracting github.com/klauspost/reedsolomon v1.9.3
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: downloading github.com/klauspost/cpuid v1.2.1
go: extracting github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/klauspost/cpuid v1.2.1
go: downloading github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: extracting github.com/miekg/dns v1.1.8
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/mattn/go-isatty v0.0.7
go: extracting github.com/djherbis/atime v1.0.0
go: downloading github.com/cheggaaa/pb v1.0.28
go: extracting github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading google.golang.org/grpc v1.20.1
go: extracting github.com/mattn/go-isatty v0.0.7
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/pkg/profile v1.3.0
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/dustin/go-humanize v1.0.0
go: extracting github.com/cheggaaa/pb v1.0.28
go: downloading github.com/gorilla/mux v1.7.0
go: extracting github.com/pkg/profile v1.3.0
go: extracting github.com/dustin/go-humanize v1.0.0
go: downloading golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: extracting github.com/gorilla/mux v1.7.0
go: downloading github.com/minio/sio v0.2.0
go: extracting google.golang.org/grpc v1.20.1
go: extracting github.com/minio/sio v0.2.0
go: downloading github.com/minio/hdfs/v3 v3.0.1
go: downloading github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: extracting github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: extracting github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: extracting golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: extracting github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: extracting golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: downloading go.opencensus.io v0.21.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/minio/hdfs/v3 v3.0.1
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/klauspost/compress v1.8.3
go: extracting go.opencensus.io v0.21.0
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/minio/lsync v1.0.1
go: extracting github.com/minio/lsync v1.0.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/klauspost/readahead v1.3.0
go: extracting github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: extracting github.com/klauspost/readahead v1.3.0
go: downloading contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: extracting contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: downloading gopkg.in/yaml.v2 v2.2.2
go: extracting gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/golang/protobuf v1.3.1
go: extracting golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: extracting github.com/golang/protobuf v1.3.1
go: extracting google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/census-instrumentation/opencensus-proto v0.2.0
go: extracting github.com/census-instrumentation/opencensus-proto v0.2.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: extracting github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: downloading github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: extracting github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: downloading github.com/nsqio/go-nsq v1.0.7
go: extracting github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: extracting github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: extracting github.com/go-sql-driver/mysql v1.4.1
go: downloading github.com/klauspost/pgzip v1.2.1
go: extracting github.com/klauspost/pgzip v1.2.1
go: downloading gopkg.in/ini.v1 v1.48.0
go: extracting gopkg.in/ini.v1 v1.48.0
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/fatih/color v1.7.0
go: extracting github.com/fatih/color v1.7.0
go: downloading cloud.google.com/go v0.37.2
go: extracting cloud.google.com/go v0.37.2
go: downloading github.com/tidwall/gjson v1.2.1
go: extracting github.com/klauspost/compress v1.8.3
go: extracting github.com/tidwall/gjson v1.2.1
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/pkg/errors v0.8.1
go: downloading git.apache.org/thrift.git v0.12.0
go: extracting github.com/pkg/errors v0.8.1
go: downloading github.com/minio/mc v0.0.0-20190924013003-643835013047
go: extracting github.com/coredns/coredns v1.4.0
go: downloading gopkg.in/Shopify/sarama.v1 v1.20.0
go: extracting github.com/minio/mc v0.0.0-20190924013003-643835013047
go: extracting gopkg.in/Shopify/sarama.v1 v1.20.0
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/mattn/go-colorable v0.1.1
go: extracting github.com/mattn/go-colorable v0.1.1
go: downloading github.com/modern-go/reflect2 v1.0.1
go: extracting github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/minio/dsync v1.0.0
go: extracting github.com/minio/dsync v1.0.0
go: downloading github.com/minio/dsync/v2 v2.0.0
go: extracting github.com/minio/dsync/v2 v2.0.0
go: downloading github.com/googleapis/gax-go v2.0.0+incompatible
go: extracting github.com/googleapis/gax-go v2.0.0+incompatible
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/davecgh/go-spew v1.1.1
go: extracting github.com/davecgh/go-spew v1.1.1
go: downloading github.com/eapache/go-resiliency v1.2.0
go: extracting github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/hashicorp/vault v1.1.0
go: extracting git.apache.org/thrift.git v0.12.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: extracting github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading github.com/pierrec/lz4 v2.0.5+incompatible
go: extracting github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: extracting google.golang.org/api v0.4.0
go: extracting github.com/pierrec/lz4 v2.0.5+incompatible
go: downloading github.com/prometheus/common v0.2.0
go: extracting github.com/prometheus/common v0.2.0
go: downloading github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: extracting github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/eapache/queue v1.1.0
go: extracting github.com/eapache/queue v1.1.0
go: downloading github.com/fatih/structs v1.1.0
go: extracting github.com/alecthomas/participle v0.2.1
go: downloading github.com/tidwall/match v1.0.1
go: extracting github.com/fatih/structs v1.1.0
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/tidwall/pretty v1.0.0
go: extracting github.com/tidwall/match v1.0.1
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/tidwall/pretty v1.0.0
go: downloading github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: extracting github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: extracting golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/rjeczalik/notify v0.9.2
go: extracting github.com/minio/sha256-simd v0.1.1
go: extracting github.com/rjeczalik/notify v0.9.2
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/lib/pq v1.0.0
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting github.com/lib/pq v1.0.0
go: downloading github.com/gogo/protobuf v1.2.1
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading github.com/nats-io/nats.go v1.8.0
go: extracting github.com/nats-io/nats.go v1.8.0
go: downloading github.com/nats-io/nuid v1.0.1
go: extracting github.com/nats-io/nuid v1.0.1
go: extracting github.com/gogo/protobuf v1.2.1
go: downloading github.com/satori/go.uuid v1.2.0
go: downloading golang.org/x/text v0.3.2
go: extracting github.com/satori/go.uuid v1.2.0
go: downloading github.com/nats-io/stan.go v0.4.5
go: extracting github.com/nats-io/stan.go v0.4.5
go: downloading golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: extracting golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: downloading github.com/marstr/guid v1.1.0
go: extracting github.com/hashicorp/vault v1.1.0
go: extracting github.com/marstr/guid v1.1.0
go: downloading github.com/nats-io/nkeys v0.0.2
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/nats-io/nkeys v0.0.2
go: downloading github.com/golang/snappy v0.0.1
go: extracting github.com/golang/snappy v0.0.1
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/hashicorp/hcl v1.0.0
go: extracting github.com/hashicorp/hcl v1.0.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting golang.org/x/text v0.3.2
go: downloading github.com/hashicorp/go-retryablehttp v0.5.3
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/hashicorp/go-retryablehttp v0.5.3
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/minio/gokrb5/v7 v7.2.5
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/minio/gokrb5/v7 v7.2.5
go: finding google.golang.org/api v0.4.0
go: finding github.com/minio/hdfs/v3 v3.0.1
go: finding github.com/minio/minio-go/v6 v6.0.37
go: finding github.com/minio/cli v1.21.0
go: finding github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/json-iterator/go v1.1.7
go: finding cloud.google.com/go v0.37.2
go: finding github.com/Azure/go-autorest v11.7.0+incompatible
go: finding github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding github.com/fatih/color v1.7.0
go: finding github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: finding github.com/minio/sha256-simd v0.1.1
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding github.com/gorilla/mux v1.7.0
go: finding github.com/mattn/go-isatty v0.0.7
go: finding github.com/minio/mc v0.0.0-20190924013003-643835013047
go: finding github.com/minio/highwayhash v1.0.0
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding google.golang.org/grpc v1.20.1
go: finding github.com/mattn/go-colorable v0.1.1
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding github.com/rjeczalik/notify v0.9.2
go: finding golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: finding golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding github.com/golang/protobuf v1.3.1
go: finding google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding gopkg.in/ini.v1 v1.48.0
go: finding golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: finding go.opencensus.io v0.21.0
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: finding github.com/gogo/protobuf v1.2.1
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding github.com/marstr/guid v1.1.0
go: finding github.com/satori/go.uuid v1.2.0
go: finding golang.org/x/text v0.3.2
go: finding github.com/census-instrumentation/opencensus-proto v0.2.0
go: finding golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: finding github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/mattn/go-runewidth v0.0.4
go: finding github.com/coreos/etcd v3.3.12+incompatible
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding github.com/gorilla/handlers v1.4.0
go: finding github.com/gorilla/rpc v1.2.0
go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: finding github.com/klauspost/compress v1.8.3
go: finding github.com/klauspost/cpuid v1.2.1
go: finding github.com/klauspost/readahead v1.3.0
go: finding github.com/klauspost/reedsolomon v1.9.3
go: finding github.com/miekg/dns v1.1.8
go: finding github.com/minio/dsync/v2 v2.0.0
go: finding github.com/minio/lsync v1.0.1
go: finding github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: finding github.com/hashicorp/vault v1.1.0
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/hashicorp/go-retryablehttp v0.5.3
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/pierrec/lz4 v2.0.5+incompatible
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding github.com/ncw/directio v1.0.5
go: finding github.com/minio/sio v0.2.0
go: finding github.com/coredns/coredns v1.4.0
go: finding github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: finding github.com/go-sql-driver/mysql v1.4.1
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/lib/pq v1.0.0
go: finding github.com/nats-io/nats.go v1.8.0
go: finding github.com/nats-io/nkeys v0.0.2
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/nats-io/stan.go v0.4.5
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: finding github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: finding gopkg.in/Shopify/sarama.v1 v1.20.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/eapache/go-resiliency v1.2.0
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding github.com/eapache/queue v1.1.0
go: finding github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: finding github.com/fatih/structs v1.1.0
go: finding github.com/tidwall/gjson v1.2.1
go: finding github.com/tidwall/match v1.0.1
go: finding github.com/tidwall/pretty v1.0.0
go: finding gopkg.in/yaml.v2 v2.2.2
go: finding github.com/klauspost/pgzip v1.2.1
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding github.com/alecthomas/participle v0.2.1
go: finding github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: finding git.apache.org/thrift.git v0.12.0
go: finding github.com/pkg/profile v1.3.0
go: finding github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: finding github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: finding github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: finding github.com/prometheus/common v0.2.0
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: finding github.com/rs/cors v1.6.0
go: finding go.uber.org/atomic v1.3.2
go: finding gopkg.in/ldap.v3 v3.0.3
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
# github.com/minio/minio/cmd
cmd/config-current.go:644:49: undefined: openid
cmd/config-current.go:645:16: undefined: openid
cmd/config-current.go:648:18: undefined: openid
cmd/config-migrate.go:2656:20: undefined: openid
cmd/globals.go:270:23: undefined: openid
cmd/sts-handlers.go:183:18: undefined: openid
cmd/sts-handlers.go:284:8: undefined: openid
cmd/sts-handlers.go:292:8: undefined: openid
make: *** [build] Error 2

```

--- a/cmd/config-current.go
+++ b/cmd/config-current.go
@@ -30,8 +30,8 @@ import (
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/event"
 	"github.com/minio/minio/pkg/event/target"
+	"github.com/minio/minio/pkg/iam/openid"
 	iampolicy "github.com/minio/minio/pkg/iam/policy"
-	"github.com/minio/minio/pkg/iam/validator"
 	xnet "github.com/minio/minio/pkg/net"
 )
 
@@ -571,7 +571,7 @@ func (s *serverConfig) loadToCachedConfigs() {
 			"Unable to populate public key from JWKS URL %s", s.OpenID.JWKS.URL)
 	}
 
-	globalIAMValidators = getAuthValidators(s)
+	globalIAMValidators = getOpenIDValidators(s)
 
 	if s.Policy.OPA.URL != nil && s.Policy.OPA.URL.String() != "" {
 		opaArgs := iampolicy.OpaArgs{
@@ -638,15 +638,15 @@ func loadConfig(objAPI ObjectLayer) error {
 	return nil
 }
 
-// getAuthValidators - returns ValidatorList which contains
+// getOpenIDValidators - returns ValidatorList which contains
 // enabled providers in server config.
 // A new authentication provider is added like below
-// * Add a new provider in pkg/iam/validator package.
-func getAuthValidators(config *serverConfig) *validator.Validators {
-	validators := validator.NewValidators()
+// * Add a new provider in pkg/iam/openid package.
+func getOpenIDValidators(config *serverConfig) *openid.Validators {
+	validators := openid.NewValidators()
 
 	if config.OpenID.JWKS.URL != nil {
-		validators.Add(validator.NewJWT(config.OpenID.JWKS))
+		validators.Add(openid.NewJWT(config.OpenID.JWKS))
 	}
 
 	return validators

--- a/cmd/config-migrate.go
+++ b/cmd/config-migrate.go
@@ -29,8 +29,8 @@ import (
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/event"
 	"github.com/minio/minio/pkg/event/target"
+	"github.com/minio/minio/pkg/iam/openid"
 	iampolicy "github.com/minio/minio/pkg/iam/policy"
-	"github.com/minio/minio/pkg/iam/validator"
 	xnet "github.com/minio/minio/pkg/net"
 	"github.com/minio/minio/pkg/quick"
 )
@@ -2654,7 +2654,7 @@ func migrateV30ToV31MinioSys(objAPI ObjectLayer) error {
 	}
 
 	cfg.Version = "31"
-	cfg.OpenID.JWKS = validator.JWKSArgs{
+	cfg.OpenID.JWKS = openid.JWKSArgs{
 		URL: &xnet.URL{},
 	}
 	cfg.Policy.OPA = iampolicy.OpaArgs{

--- a/cmd/config-versions.go
+++ b/cmd/config-versions.go
@@ -22,8 +22,8 @@ import (
 	"github.com/minio/minio/cmd/crypto"
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/event/target"
+	"github.com/minio/minio/pkg/iam/openid"
 	iampolicy "github.com/minio/minio/pkg/iam/policy"
-	"github.com/minio/minio/pkg/iam/validator"
 	"github.com/minio/minio/pkg/quick"
 )
 
@@ -817,7 +817,7 @@ type serverConfigV31 struct {
 	// OpenID configuration
 	OpenID struct {
 		// JWKS validator config.
-		JWKS validator.JWKSArgs `json:"jwks"`
+		JWKS openid.JWKSArgs `json:"jwks"`
 	} `json:"openid"`
 
 	// External policy enforcements.
@@ -872,7 +872,7 @@ type serverConfigV32 struct {
 	// OpenID configuration
 	OpenID struct {
 		// JWKS validator config.
-		JWKS validator.JWKSArgs `json:"jwks"`
+		JWKS openid.JWKSArgs `json:"jwks"`
 	} `json:"openid"`
 
 	// External policy enforcements.
@@ -916,7 +916,7 @@ type serverConfigV33 struct {
 	// OpenID configuration
 	OpenID struct {
 		// JWKS validator config.
-		JWKS validator.JWKSArgs `json:"jwks"`
+		JWKS openid.JWKSArgs `json:"jwks"`
 	} `json:"openid"`
 
 	// External policy enforcements.

--- a/cmd/globals.go
+++ b/cmd/globals.go
@@ -33,8 +33,8 @@ import (
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/certs"
 	"github.com/minio/minio/pkg/dns"
+	"github.com/minio/minio/pkg/iam/openid"
 	iampolicy "github.com/minio/minio/pkg/iam/policy"
-	"github.com/minio/minio/pkg/iam/validator"
 	"github.com/minio/minio/pkg/pubsub"
 )
 
@@ -268,7 +268,7 @@ var (
 	standardExcludeCompressContentTypes = []string{"video/*", "audio/*", "application/zip", "application/x-gzip", "application/x-zip-compressed", " application/x-compress", "application/x-spoon"}
 
 	// Authorization validators list.
-	globalIAMValidators *validator.Validators
+	globalIAMValidators *openid.Validators
 
 	// OPA policy system.
 	globalPolicyOPA *iampolicy.Opa

--- a/cmd/signature-v4-utils_test.go
+++ b/cmd/signature-v4-utils_test.go
@@ -17,9 +17,10 @@
 package cmd
 
 import (
-	"github.com/minio/minio/cmd/crypto"
 	"net/http"
 	"testing"
+
+	"github.com/minio/minio/cmd/crypto"
 )
 
 // TestSkipContentSha256Cksum - Test validate the logic which decides whether

--- a/cmd/sts-handlers.go
+++ b/cmd/sts-handlers.go
@@ -27,8 +27,8 @@ import (
 	xhttp "github.com/minio/minio/cmd/http"
 	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/pkg/auth"
+	"github.com/minio/minio/pkg/iam/openid"
 	iampolicy "github.com/minio/minio/pkg/iam/policy"
-	"github.com/minio/minio/pkg/iam/validator"
 	"github.com/minio/minio/pkg/wildcard"
 	ldap "gopkg.in/ldap.v3"
 )
@@ -181,7 +181,7 @@ func (sts *stsAPIHandlers) AssumeRole(w http.ResponseWriter, r *http.Request) {
 
 	var err error
 	m := make(map[string]interface{})
-	m["exp"], err = validator.GetDefaultExpiration(r.Form.Get("DurationSeconds"))
+	m["exp"], err = openid.GetDefaultExpiration(r.Form.Get("DurationSeconds"))
 	if err != nil {
 		writeSTSErrorResponse(ctx, w, ErrSTSInvalidParameterValue, err)
 		return
@@ -282,7 +282,7 @@ func (sts *stsAPIHandlers) AssumeRoleWithJWT(w http.ResponseWriter, r *http.Requ
 	m, err := v.Validate(token, r.Form.Get("DurationSeconds"))
 	if err != nil {
 		switch err {
-		case validator.ErrTokenExpired:
+		case openid.ErrTokenExpired:
 			switch action {
 			case clientGrants:
 				writeSTSErrorResponse(ctx, w, ErrSTSClientGrantsExpiredToken, err)
@@ -290,7 +290,7 @@ func (sts *stsAPIHandlers) AssumeRoleWithJWT(w http.ResponseWriter, r *http.Requ
 				writeSTSErrorResponse(ctx, w, ErrSTSWebIdentityExpiredToken, err)
 			}
 			return
-		case validator.ErrInvalidDuration:
+		case openid.ErrInvalidDuration:
 			writeSTSErrorResponse(ctx, w, ErrSTSInvalidParameterValue, err)
 			return
 		}

--- a/pkg/iam/openid/jwks.go
+++ b/pkg/iam/openid/jwks.go
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package validator
+package openid
 
 import (
 	"crypto"

--- a/pkg/iam/openid/jwks_test.go
+++ b/pkg/iam/openid/jwks_test.go
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package validator
+package openid
 
 import (
 	"bytes"

--- a/pkg/iam/openid/jwt.go
+++ b/pkg/iam/openid/jwt.go
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package validator
+package openid
 
 import (
 	"crypto"

--- a/pkg/iam/openid/jwt_test.go
+++ b/pkg/iam/openid/jwt_test.go
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package validator
+package openid
 
 import (
 	"crypto"

--- a/pkg/iam/openid/validators.go
+++ b/pkg/iam/openid/validators.go
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package validator
+package openid
 
 import (
 	"errors"

--- a/pkg/iam/openid/validators_test.go
+++ b/pkg/iam/openid/validators_test.go
@@ -14,10 +14,14 @@
  * limitations under the License.
  */
 
-package validator
+package openid
 
 import (
+	"net/http"
+	"net/http/httptest"
 	"testing"
+
+	xnet "github.com/minio/minio/pkg/net"
 )
 
 type errorValidator struct{}
@@ -31,7 +35,32 @@ func (e errorValidator) ID() ID {
 }
 
 func TestValidators(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("content-type", "application/json")
+		w.Write([]byte(`{
+  "keys" : [ {
+    "kty" : "RSA",
+    "kid" : "1438289820780",
+    "use" : "sig",
+    "alg" : "RS256",
+    "n" : "idWPro_QiAFOdMsJD163lcDIPogOwXogRo3Pct2MMyeE2GAGqV20Sc8QUbuLDfPl-7Hi9IfFOz--JY6QL5l92eV-GJXkTmidUEooZxIZSp3ghRxLCqlyHeF5LuuM5LPRFDeF4YWFQT_D2eNo_w95g6qYSeOwOwGIfaHa2RMPcQAiM6LX4ot-Z7Po9z0_3ztFa02m3xejEFr2rLRqhFl3FZJaNnwTUk6an6XYsunxMk3Ya3lRaKJReeXeFtfTpShgtPiAl7lIfLJH9h26h2OAlww531DpxHSm1gKXn6bjB0NTC55vJKft4wXoc_0xKZhnWmjQE8d9xE8e1Z3Ll1LYbw",
+    "e" : "AQAB"
+  }, {
+    "kty" : "RSA",
+    "kid" : "1438289856256",
+    "use" : "sig",
+    "alg" : "RS256",
+    "n" : "zo5cKcbFECeiH8eGx2D-DsFSpjSKbTVlXD6uL5JAy9rYIv7eYEP6vrKeX-x1z70yEdvgk9xbf9alc8siDfAz3rLCknqlqL7XGVAQL0ZP63UceDmD60LHOzMrx4eR6p49B3rxFfjvX2SWSV3-1H6XNyLk_ALbG6bGCFGuWBQzPJB4LMKCrOFq-6jtRKOKWBXYgkYkaYs5dG-3e2ULbq-y2RdgxYh464y_-MuxDQfvUgP787XKfcXP_XjJZvyuOEANjVyJYZSOyhHUlSGJapQ8ztHdF-swsnf7YkePJ2eR9fynWV2ZoMaXOdidgZtGTa4R1Z4BgH2C0hKJiqRy9fB7Gw",
+    "e" : "AQAB"
+  } ]
+}
+`))
+		w.(http.Flusher).Flush()
+	}))
+	defer ts.Close()
+
 	vrs := NewValidators()
+
 	if err := vrs.Add(&errorValidator{}); err != nil {
 		t.Fatal(err)
 	}
@@ -58,7 +87,18 @@ func TestValidators(t *testing.T) {
 		t.Fatalf("Unexpected number of vids %v", vids)
 	}
 
-	if vids[0] != "err" {
-		t.Fatalf("Unexpected vid %v", vids[0])
+	u, err := xnet.ParseURL(ts.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = vrs.Add(NewJWT(JWKSArgs{
+		URL: u,
+	})); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err = vrs.Get("jwt"); err != nil {
+		t.Fatal(err)
 	}
 }

